### PR TITLE
Use semantic controls throughout chat

### PR DESCRIPTION
**Pull request by sachaservan**
## Summary
- render persistent chat, project, and new-chat destinations as links with native modified-click behavior
- use native buttons for fixed-context actions, remove nested interactive controls, and improve accessible labels
- centralize chat route and link-click helpers and add semantic control tests

## Validation
- `npm run lint`
- `npx tsc --noEmit`
- `npx vitest run` (1,154 tests passed)

Playwright was not run because its configuration starts the development server, which project instructions prohibit.

---
## Summary by cubic
Standardized navigation and actions across chat and project UIs using real links and buttons to improve accessibility and preserve native modified-click behavior. New chat and chat links now carry storage context (local/cloud) and project across tabs, with centralized path helpers.

- **Refactors**
  - Added `utils/navigation` with `getChatPath`, `getNewChatPath`, and `isPlainPrimaryClick`; updated `use-chat-router` and UI to use them.
  - New chat controls are links that preserve local/cloud and project context; plain clicks run in-app handlers.
  - `ChatListItem` supports `href`; `ChatList` accepts `getChatHref` to supply destinations.
  - Projects list items link to project new chat (encrypted projects render non-links with a delete button).
  - `newchat` reads `?storage=local`; `ChatInterface` passes `initialNewChatIsLocalOnly` and storage selects the correct blank chat. Converted several controls to semantic buttons/links (Artifact card, Document list, Chats/Projects headers, project rename, settings/close).

- **Bug Fixes**
  - Default `Button` uses `type="button"`; added missing `type="button"` on action buttons.
  - Preserved modified- and middle-click behavior for links; marked current new chat with `aria-current`.
  - Routed local new chats stay selected after storage loads; new chat context persists across tabs/windows.
  - Accessibility: selection popover uses `role="toolbar"`, avoided anchors without `href`, added labels for close/settings.
  - Tests: navigation helpers, chat list link semantics, `newchat` local routing, button default type, and storage selection after reload.

<sup>Written for commit ef377aa59ffd7297de25d2ff3c5ccd85789600b3. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/tinfoilsh/tinfoil-webapp/pull/450?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/components/project/project-sidebar.tsx [hunk 1]
```diff
@@ -744,18 +714,21 @@ export function ProjectSidebar({
           <div className="flex flex-col items-center gap-1 px-2">
             {/* New chat button */}
             <div className="group relative">
-              <button
-                type="button"
-                onClick={(e) => handleNewChatButtonClick(e, onNewChat)}
-                onAuxClick={handleNewChatAuxClick}
+              <Link
```
[the remark this answers is not in the document]
Comment (sachaservan) [reply src/components/project/project-sidebar.tsx:720]: Not extracting this. The semantic decision is already centralized in `isPlainPrimaryClick`; wrapping the remaining three local statements would obscure the click flow without meaningful reuse.

File: tests/pages/newchat.test.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+describe('NewChatPage', () => {
+  beforeEach(() => {
+    vi.clearAllMocks()
+    mocks.router.isReady = true
+    mocks.router.query = {}
+  })
+
+  it('starts a routed local new chat in local-only mode', () => {
+    mocks.router.query = { storage: 'local' }
```
[the remark this answers is not in the document]
Comment (sachaservan) [reply tests/pages/newchat.test.tsx:37]: Keeping these literals is intentional: this test pins the public `storage=local` URL contract. Importing the implementation constants would allow an accidental contract change to pass; the navigation utility tests cover generation separately.

File: tests/pages/newchat.test.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    render(<NewChatPage />)
+
+    expect(mocks.chatInterface).toHaveBeenCalledWith(
+      expect.objectContaining({ initialNewChatIsLocalOnly: true }),
+    )
+  })
+
+  it('uses cloud mode for the default new-chat destination', () => {
```
[the remark this answers is not in the document]
Comment (sachaservan) [reply tests/pages/newchat.test.tsx:46]: Not adding this test: the not-ready path is a trivial `null` guard with no user behavior. The meaningful ready-state routing paths are already covered.
